### ISSUE TITLE
[Bug] Confidence scores capped at ~0.85; Precision/Recall collapse above this threshold

### ISSUE THREAD

**Issue by gutembergandrade** [roboflow/rf-detr#526]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I am experiencing a potential issue with score calibration or logit constraints in RF-DETR. When plotting Precision and Recall against the confidence threshold, both metrics drop abruptly to zero at approximately 0.85.

Observations:

* Global Max Score: I manually verified the prediction tensors, and the maximum confidence score across the entire validation set is 0.855288.

* The "Cliff": As shown in the attached PR-Threshold curve, the model performs well up to 0.8 around 0.8 confidence, but it is physically incapable of producing a prediction with higher confidence.

* Precision Behavior: Once the threshold exceeds 0.855, Precision drops to 0.0, indicating that the evaluation logic is likely penalizing the absence of predictions rather than treating the metric as undefined.

* EMA Weights: I am using Standard weights. But the problem persists in every version of RF-DETR trained with this dataset, including the EMA version.

Environment:

Model: RF-DETR Large

Dataset: Custom warehouse dataset

Training Specs: 36 epochs (with early stopping at 25 epochs, best epoch was 15)

Screenshot: <img width="779" height="454" alt="Image" src="https://github.com/user-attachments/assets/87c87800-6efb-4b13-830b-f2a4c65df76d" />

### Environment

- RF-DETR version: https://github.com/lab176344/rf-detr/tree/feature/mlflow-integeration
- OS: Ubuntu 22.04 LTS
- Python version: 3.13
- PyTorch version: 1.13.0+cu116
- CUDA/cuDNN version: 11.5.119
- GPU/CPU hardware: Tesla A100 80GB on Google Colab (although the problem persists in every env we tested) 

### Minimal Reproducible Example

```python
from rfdetr import RFDETRLarge

model = RFDETRLarge()

dataset_path = "/rf_detr_dataset"
output_path = "/roboflow-rfdetr"

model.train(
    dataset_dir=dataset_path,
    epochs=36,
    batch_size=16,
    grad_accum_steps=1,
    lr=1e-4,
    output_dir=output_path,
    early_stopping=True
)
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob-roboflow** [comment]
A higher confidence doesn't make for a better model. Why is it a problem if the model doesn't predict more than 0.85 confidence?
